api/task/cluster.go

```diff
@@ -124,26 +125,44 @@ func ClusterDataset(datasetSource metadata.DatasetSource, schemaFile string, ind
 }
 
 // Cluster will cluster the dataset fields using a primitive.
-func Cluster(datasetInputDir string, dataset string, variable string, features []*model.Variable) ([]*ClusterPoint, error) {
-	step, err := description.CreateSlothPipeline("time series clustering",
-		"k-means time series clustering", "", "", features)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to create sloth pipeline")
+func Cluster(datasetInputDir string, dataset string, variable string, features []*model.Variable) (bool, []*ClusterPoint, error) {
+	var clusteringVar *model.Variable
+	for _, v := range features {
+		if v.Name == variable {
+			clusteringVar = v
+		}
+	}
+
+	var step *pipeline.PipelineDescription
+	var err error
+	addMeta := false
+	if model.IsImage(clusteringVar.Type) {
+		step, err = description.CreateImageClusteringPipeline("business", "basic image clustering", []*model.Variable{clusteringVar})
+		if err != nil {
+			return false, nil, errors.Wrap(err, "unable to create image clustering pipeline")
```

Comment:
We probably don't need to call `errors.Wrap` on most of these errors.  Since its our code, we should already be creating the error using `errors.Errorf` further down the call stack.  Adding the `Wrap` just generates more log noise.  It looks like the pre-existing code was wrapping when it shouldn't have been as well.